ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -318,29 +325,40 @@ private EndpointResponse handlePullQuery(
         && isInternalRequest.orElse(true);
 
     // Only check the rate limit at the forwarding host
+    Decrementer decrementer = null;
     if (!isAlreadyForwarded) {
       PullQueryExecutionUtil.checkRateLimit(rateLimiter);
+      decrementer = concurrencyLimiter.increment();
```

Comment:
I'm wondering if we should check the concurrent limit at the final host instead of the forwarding host?

Reply:
Hmm, that's a good point. 

My concern is that by only limiting on the forwarding host, the final hosts may already exceed its concurrency quota while the forwarding host does not. But since today all table scans are full scans, by setting this limit on a HA cluster to `N` per-node we are effectively allowing the concurrent requests to be `3N` anyways assuming the num.partitions would usually > 3 so each node would host at least some partitions of each table. So please ignore this comment.

Reply:
Yes, I agree with the partial result comment.  This isn't a perfect check because the forwarded node may be unlucky and have more concurrent requests than this limit, but if partitions are uniformly distributed (which they will be on average), then they should be fairly equivalent.  And for the reasons above, it's much easier to do the check here.